opm/output/eclipse/summaryReader.cpp

```diff
@@ -0,0 +1,333 @@
+/*
+  Copyright 2016 Statoil ASA.
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <opm/output/eclipse/summaryReader.hpp>
+
+
+
+
+bool SummaryReader::open(const char* smspecFile1, const stringlist_type* unsmryFile1, const char * smspecFile2, const stringlist_type* unsmryFile2){
+  //opens the files
+  
+  
+  ecl_sum1 = ecl_sum_fread_alloc(smspecFile1, unsmryFile1, ":");
+  ecl_sum2 = ecl_sum_fread_alloc(smspecFile2, unsmryFile2, ":");
+  
+
+  if (ecl_sum1 == nullptr || ecl_sum2 == nullptr) {
+    OPM_THROW(std::runtime_error, "Not able to open files");
+  }
+  else {
+    return true;
+  }
+}
+
+void SummaryReader::close() {
+  ecl_sum_free_data(ecl_sum1);
+  ecl_sum_free_data(ecl_sum2);
+  stringlist_free(keys1);
```

Comment:
these are not guaranteed to be allocated. stringlist_free does not check for a nullptr. if()'s required.